internal/pkg/v2/infrastructure/redis/reading.go

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package redis
+
+import (
+	"encoding/json"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/common"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	model "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/gomodule/redigo/redis"
+
+	"github.com/google/uuid"
+)
+
+const ReadingsCollection = "v2:reading"
+
+var emptyBinaryValue = make([]byte, 0)
+
+// Add a reading to the database
+func addReading(conn redis.Conn, r model.Reading) (id string, err error) {
+	var m []byte
+	var baseReading *model.BaseReading
+	switch newReading := r.(type) {
+	case model.BinaryReading:
+		// Clear the binary data since we do not want to persist binary data to save on memory.
+		newReading.BinaryValue = emptyBinaryValue
+
+		baseReading = &newReading.BaseReading
+		checkReadingValue(baseReading)
+		m, err = json.Marshal(newReading)
```

Comment:
These are duplicated and can be moved below the switch.  Leaving `case model.SimpleReading:` empty.

Reply:
It won't work if the above code is moved out of the switch block since the `newReading` value should be passed as a struct.

Reply:
Ah, I see it now; we need those lines in the switch as that is the small block where we know which concrete type the interface is carrying. Which is the point @lindseysimple was trying to convey earlier. Apologies for my being slow on the uptake.

I did notice the update to `model.Reading` and it got me thinking we can take advantage of closure to perhaps simplify the overall code. I [created a gist](https://gist.github.com/andresrinivasan/0d7429d685008d2f6ed9901aab3183da#file-reading-go) after playing with the code. Outside of any errors I may have created, we are straying into philosophy so please consider this as you would any form of artistic expression.

Reply:
Ok, i see now that `newReading` is only in scope of switch. THX!